src/Uno.UI.Composition/Composition/Visual.skia.cs

```diff
@@ -16,6 +15,49 @@ public partial class Visual : global::Microsoft.UI.Composition.CompositionObject
 	private CompositionClip? _clip;
 	private Vector2 _anchorPoint = Vector2.Zero; // Backing for scroll offsets
 	private int _zIndex;
+	private bool _matrixDirty = true;
+	private SKMatrix _totalMatrix = SKMatrix.Identity;
+
+	/// <returns>true if wasn't dirty</returns>
+	protected internal virtual bool SetMatrixDirty() => _matrixDirty = true;
+
+	/// <summary>
+	/// This is the final transformation matrix from the origin for this Visual.
+	/// </summary>
+	[DebuggerDisplay("{TotalMatrixString}")]
+	internal SKMatrix TotalMatrix
```

Comment:
Why have a `SKMatrix` here ? I would use a `Matrix4x4` instead.

Reply:
Done. I just remembered why: Matrix4x4 uses the weird column-major style that contradicts the convention everyone else uses for no good reason 🤦